src/hashTable.js

```diff
@@ -1,63 +1,94 @@
 class Node{
-  constructor() {
-    this.key = 0,
-    this.data = 0,
+  constructor(key, data) {
+    this.key = key,
+    this.data = data,
     this.next = null
   }
 }
 
 export default class Hashtable{
   constructor(){
-    this._length,
+    this._length = 0,
     this.dataStore = []
   }
 
-  hashFunction(key) {
+  //takes a key and returns a hash number which will be where the hash will live.
+  hashFunction(key){
     let sum = 0
     for (let i = 0; i < key.length; i++) {
       sum += key.charCodeAt(i)
     }
     return sum % 31
   }
+  //get the number of hashes in your hash table.
+  size(){ return this._length }
 
-  chain(){
-    //iterate through the linkedlist until you reach a node whos next is null.
-    //set that next to the new node that is created.
-  }
 
-<<<<<<< Updated upstream
-  put(id, data){
-    let hash = hashFunction(id)
-=======
+  //creates a new hash or chains off an existing hash.
   put(key, data){
+    let newNode = new Node(key, data)
     let hash = this.hashFunction(key)
->>>>>>> Stashed changes
     //see if hash # exists on the hash table.
-    if(!this.dataStore.contains(hash)){
+    if(!this.dataStore[hash]){
       //if it doesnt, create a new hash index in the table and put data there.
-
+      this.dataStore[hash] = newNode
     }else{
       //otherwise chain off the lead hash into a linkedlist.
-<<<<<<< Updated upstream
-      
-=======
-
->>>>>>> Stashed changes
+      this.chain(key, data, hash, newNode)
     }
+    ++this._length
   }
 
-  get(){
+  get(key){
+    let hash = this.hashFunction(key)
     //finds the indexed hash number and looks at the key to locate the correct node.
-    //returns that node's data.
+    if(this.dataStore[hash]){
+      let currentNode = this.dataStore[hash]
+      if(currentNode.key === key){
+        //returns that node's data.
+        return currentNode.data
+      }else{
+        while(currentNode.next){
```

Comment:
I prefer the use of explicit comparison, for clarity, and to avoid relying on truthy and falsey values:
```js
while( currentNode.next !== null ) {
```